src/inc/til/point.h

```diff
@@ -3,403 +3,264 @@
+                throw gsl::narrowing_error{};
+            }
+            return val;
         }
+    }
 
-        constexpr point(ptrdiff_t x, ptrdiff_t y) noexcept :
-            _x(x),
-            _y(y)
-        {
-        }
+    struct point
+    {
+        CoordType x = 0;
+        CoordType y = 0;
 
-        // This template will convert to size from anything that has an X and a Y field that appear convertible to an integer value
-        template<typename TOther>
-        constexpr point(const TOther& other, std::enable_if_t<std::is_integral_v<decltype(std::declval<TOther>().X)> && std::is_integral_v<decltype(std::declval<TOther>().Y)>, int> /*sentinel*/ = 0) :
-            point(static_cast<ptrdiff_t>(other.X), static_cast<ptrdiff_t>(other.Y))
-        {
-        }
+        constexpr point() noexcept = default;
 
-        // This template will convert to size from anything that has a x and a y field that appear convertible to an integer value
-        template<typename TOther>
-        constexpr point(const TOther& other, std::enable_if_t<std::is_integral_v<decltype(std::declval<TOther>().x)> && std::is_integral_v<decltype(std::declval<TOther>().y)>, int> /*sentinel*/ = 0) :
-            point(static_cast<ptrdiff_t>(other.x), static_cast<ptrdiff_t>(other.y))
+        constexpr point(CoordType x, CoordType y) noexcept :
+            x{ x }, y{ y }
         {
         }
 
         // This template will convert to point from floating-point args;
         // a math type is required. If you _don't_ provide one, you're going to
         // get a compile-time error about "cannot convert from initializer-list to til::point"
-        template<typename TilMath, typename TOther>
-        constexpr point(TilMath, const TOther& x, const TOther& y, std::enable_if_t<std::is_floating_point_v<TOther>, int> /*sentinel*/ = 0) :
-            point(TilMath::template cast<ptrdiff_t>(x), TilMath::template cast<ptrdiff_t>(y))
+        template<typename TilMath, typename T>
+        constexpr point(TilMath, const T x, const T y) :
+            x{ TilMath::template cast<CoordType>(x) }, y{ TilMath::template cast<CoordType>(y) }
         {
         }
 
-        // This template will convert to size from anything that has a X and a Y field that are floating-point;
-        // a math type is required. If you _don't_ provide one, you're going to
-        // get a compile-time error about "cannot convert from initializer-list to til::point"
-        template<typename TilMath, typename TOther>
-        constexpr point(TilMath, const TOther& other, std::enable_if_t<std::is_floating_point_v<decltype(std::declval<TOther>().X)> && std::is_floating_point_v<decltype(std::declval<TOther>().Y)>, int> /*sentinel*/ = 0) :
-            point(TilMath::template cast<ptrdiff_t>(other.X), TilMath::template cast<ptrdiff_t>(other.Y))
+        constexpr bool operator==(const point rhs) const noexcept
         {
+            return __builtin_memcmp(this, &rhs, sizeof(rhs)) == 0;

```

Comment:
Why the private `__builtin_memcmp` instead of the public `memcmp`?

Reply:
(leave comment at least...)

Reply:
I added the following comment to all 3 files:
```cpp
// `__builtin_memcmp` isn't an official standard, but it's the
// only way at the time of writing to get a constexpr `memcmp`.
```

See [`P1944R1`](http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2020/p1944r1.pdf), section III B.

Reply:
Blah. OK. Thank you.